packages/fuels/tests/scripts.rs

```diff
@@ -15,7 +15,7 @@ async fn test_transaction_script_workflow() -> Result<(), Error> {
     let provider = wallet.get_provider()?;
     let receipts = execution_script.execute(provider).await?;
 
-    let response = call_handler.get_response(receipts)?;
+    let response = call_handler.get_response(receipts, execution_script.tx.id())?;
```

Comment:
Feels a bit off to pass the tx id here only so that it can be included in the response struct. Suddenly the users needs to know about the internals of 2 layers.
We could call  `get_executable_call()` and extract the id from within `get_response` but this is unreliable since we have no guarantee that the user won't edit the transaction before executing. Maybe caching the underlying transaction and giving the user a mut ref would solve this.

Reply:
Ha! I actually had the same nit but brushed it off. 

Now that someone else has also thought of that... that's enough confirmation to me that we should do the way @MujkicA suggested.

`get_executable_call()` is callable from within `get_response`, so... why not? Less work for the user.

Reply:
I do confer with @iqdecay that hitting the network in `get_response` feels unnecessary, but passing `tx_id` here is still wonky. I'm exploring memoizing/caching the transaction after execution and providing it through `get_response` in an impossible-to-misuse manner.